core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -12098,20 +12101,23 @@ public String getUserNameFromUserID(String userID) throws UserStoreException {
             return ((AbstractUserStoreManager) userStore.getUserStoreManager())
                     .getUserNameFromUserID(userStore.getDomainFreeUserId());
         }
+
+        if (isUniqueUserIdEnabledInUserStore(userStore)) {
+            return getUserNameFromCurrentUserStore(userID, userStore);
+        } else {
+            return userUniqueIDManger.getUser(userID, this).getDomainQualifiedUsername();
+        }
+    }
+
+    private String getUserNameFromCurrentUserStore(String userID, UserStore userStore) throws UserStoreException {
```

Comment:
Need a method comment, explaining what it supposed to do.

Reply:
done